pyproject.toml

```
@@ -1,3 +1,7 @@
+[project]
+name = "blake3"
+version = "1.0.4"
```

we already have the version hardcoded [here](https://github.com/oconnor663/blake3-py/blob/1.0.4/Cargo.toml#L8), is there a possibility to keep only one hard-coded version in the repo?

---

for instance datafusion has [this](https://github.com/apache/datafusion-python/blob/46.0.0/pyproject.toml#L47)

---

or granian [even better](https://github.com/emmett-framework/granian/blob/v2.3.0/pyproject.toml#L26-L32)?

---

actually, it looks like the ones added in granian are redundant, looking at the datafusion wheels.
```suggestion
dynamic = ["version"]
```